texture.cpp

```diff
@@ -1,16 +1,22 @@
 #include "texture.h"
 
+#include "renderutils.h"
+
+// Ignore numerous warnings from stb_image.h
 #pragma warning(push, 0)
 #define STB_IMAGE_IMPLEMENTATION
 #include "stb_image.h"
 #pragma warning(pop)
 
 #include <iostream>
 
-#include "renderutils.h"
+
+Texture::Texture()
+{
+}
```

Comment:
Empty initializer can be ditched. It will just automatically be generated by the compiler I believe. Alternatively you can declare it as default in the header.

Reply:
So actually I was getting some problems with my use of SolidColorTexture. However I have it defined inheriting from Texture it tries to call the default constructor and gives an error
```
error C2512: 'Texture': no appropriate default constructor available
```

Doing this `Texture("")` in the SolidColorTexture constructor alleviates it, but I don't want to call the Texture constructor in this way because it does a lot extra. (I do need to reorganize how I have these classes setup though). 
```
SolidColorTexture::SolidColorTexture(unsigned char r, unsigned char g, unsigned char b, unsigned char a)
    : Texture("")
{
```